freyja/utils.py

```diff
@@ -54,6 +120,43 @@ def prepLineageDict(agg_d0, thresh=0.001):
     for line in agg_d0.index:
         counter_summary += Counter(agg_d0.loc[line, 'linDict'])
 
+    # Aggregate the lineages specified in config file into a single
+    # key/value pair.
+    if config is not None:
+        # Returns the first name where the queried lineage is present in the
+        # config memebers.
+        for i, i_config in config.items():
+            for lin in i_config['members']:
+                if '*' in lin:
+                    config[i]['members'].extend(lineage_info
+                                                [lin.replace('*', '')]
+                                                ['children']
+                                                )
+        # aggregate all the members into 1ist
+        config_members = [lin for val in config.values() for lin in val[
+                          'members']]
+        # Aggregate lineages into a single dict key/value pair
+        for j, sampLabel in enumerate(agg_d0.index):
+            # This list will contain the keys that have been created
+            # in the dict. This is used to ensure that the same key is
+            # not created twice or that value of the key is doubled.
+            created_keys = []
+            linDictMod = copy.deepcopy(agg_d0.loc[sampLabel, 'linDict'])
+            linDictModCopy = copy.deepcopy(
+                agg_d0.loc[sampLabel, 'linDict'])
+            for rInd in linDictModCopy.keys():
+                if rInd in config_members:
+                    if get_name(rInd, config) in created_keys:
+                        linDictMod[
+                            get_name(rInd, config)
+                            ] += linDictMod.pop(rInd)
+                    else:
+                        linDictMod[
+                            get_name(rInd, config)
+                            ] = linDictMod.pop(rInd)
+                        created_keys.append(get_name(rInd, config))
+            agg_d0.loc[sampLabel, 'linDict'] = [linDictMod]
```

Comment:
why is linDictMod being placed inside of a list? If we want to do this, we'll have to modify below: 

```  linDictMod = copy.deepcopy(agg_d0.loc[sampLabel, 'linDict'])[0]
        linDictModCopy = copy.deepcopy(agg_d0.loc[sampLabel, 'linDict'])[0]```

Reply:
this change  seems to have let everything work now, but I don't get the groupings specified it the plot_config.yaml file?

Reply:
I changed `agg_d0.loc[sampLabel, 'linDict'] = [linDictMod]` to replace all the `lindict` values at once towards the end using `agg_d0.loc[:, 'linDict'] = processed_linDictMod`